File: 2981/Solution.java
class Solution {
    public int maximumLength(String s) {
        int[][] a = new int[26][3];
        int n = s.length();
        int i = 0;
        while(i < n) {
            char c = s.charAt(i);
            int start = i++;
            while(i < n && c == s.charAt(i)) i++;
            Arrays.sort(a[c - 'a']);
            a[c-'a'][0] = Math.max(a[c-'a'][0], i - start); 
        }
        int res = 0;
        for(int[] r : a){
            Arrays.sort(r);
            int i1 = r[0], i2 = r[1], i3 = r[2];
            if(i1 == i2 && i1 == i3) res = Math.max(res, i1);
            if(i3 == i2 + 1 || i3 == i2) res = Math.max(res, i3 - 1);
            else res = Math.max(res, i3 - 2);
        }
        return res == 0 ? -1 : res;
    }
}

File: 2981/Solution.go
func maximumLength(s string) int {
    i, n := 0, len(s)
    arr := make([][]int, 26)
    for j := range arr {
        arr[j] = make([]int, 3)
    }
    for i < n {
        cur := s[i]
        start := i
        i++ 
        for i < n && cur == s[i] { i++ }
        sort.Ints(arr[cur - 'a'])
        arr[cur - 'a'][0] = max(arr[cur - 'a'][0], i - start)
    }
    res := 0
    for _, r := range arr {
        sort.Ints(r)
        a, b, c := r[0], r[1], r[2]
        if a == b && a == c { res = max(res, a) }

        if c == b + 1 || c == b { 
            res = max(res, c - 1)
        } else { res = max(res, c - 2) }
    }
    if res == 0 {
        return  -1
    }
    return res
}
